proton_driver/columns/mapcolumn.py

```diff
@@ -51,7 +54,9 @@ def write_items(self, items, buf):
 
 
 def create_map_column(spec, column_by_spec_getter):
-    key, value = spec[4:-1].split(',')
+    # Match commas outside of parentheses, so we don't match the comma in
+    # Decimal types.
+    key, value = comma_re.split(spec[4:-1])
```

Comment:
what if 'map(string, map(string, int))' ?

Reply:
This regex only matches commas that are not enclosed in left and right parentheses.So `map(string, map(string, int))` will be splited to `string` and `map(string,int)` correctly.

Reply:
Regex may not be capable to handle recursive things like below. For extreme case like below,

<img width="667" alt="image" src="https://github.com/user-attachments/assets/182153e4-95e5-4f02-a6ed-9e41df29444e">

Reply:
use the util func in json pr?